### PR TITLE
feat: Expand camera control ability

### PR DESCRIPTION
**Pull request by mryel00**
This will expand the ability to control the cameras.
Current method is limited to a few key controls that are controlled with cli arguments.
The new methods will add http, cli parsing and file parsing options to better control the camera at runtime and start.

- http: `http://<ip-address>:<port>/<control_to_change>=<value>?<type>`
- cli parsing: `-c {<control_to_change1>: {'value': <value1>, 'type': <type1>}, <control_to_change2>: {'value': <value2>, 'type': <type2>}}`
- file parsing: `-cf <path-to-file>` (file containing dictionary structure like cli parsing)

The type parameter is needed to convert the string coming from the input to the correct type without getting an error.
This might could get improved.

### CODE REVIEW

File: spyglass/cli.py [hunk 1]
```diff
@@ -38,11 +38,13 @@ def main(args=None):
         parsed_args.lensposition,
         parse_autofocus_speed(parsed_args.autofocusspeed))
 
+    print('Available controls:\n'+str(picam2.camera_controls))
```
Comment (roamingthings) [inline spyglass/cli.py:41]: Would it make sense to at an option to list all available controls and then exit? I don't think we need to list them each time spyglass starts

**Comment by roamingthings** [comment]
Regarding the type field I wonder if it would make sense to implement something that 'knows' the type of each control. Maybe it's even possible to parse the values using Python itself.

**Comment by roamingthings** [comment]
I've created #22 as a feature request so we can discuss how we want this feature do work

File: spyglass/cli.py (hunk 1 above)
Comment (mryel00) [reply spyglass/cli.py:41]: ofc we should add a cli argument. This was only a quick dirty way on a WIP branch to quickly see all of the controls.

**Comment by mryel00** [comment]
Added automatic type conversion for http, support for control inputs through cli and an option to use controls through the service config.
cli:
`-c '{"<control_to_change1>": <value1>, "<control_to_change2>": <value2>}'`
Service config:
`CONTROLS="{'<control_to_change1>': <value1>, '<control_to_change2>': <value2>}"`
These will change the controls directly at startup and they can still be changed through the http method.

The README isn't udpated yet and some information about the controls should be added to the standard service config.
Also there is still missing a cli command to show the available camera_controls.

**Review by roamingthings — APPROVED** [review]
Looks good. Thank you for the contribution.

**Pull request merged**